pkg/kubelet/volumemanager/reconciler/reconstruct_common.go

```diff
@@ -291,16 +286,6 @@ func (rc *reconciler) reconstructVolume(volume podVolume) (*reconstructedVolume,
 		}
 	}
 
-	// Check existence of mount point for filesystem volume or symbolic link for block volume
-	isExist, checkErr := rc.operationExecutor.CheckVolumeExistenceOperation(volumeSpec, checkPath, volumeSpec.Name(), rc.mounter, uniqueVolumeName, volume.podName, pod.UID, attachablePlugin)
-	if checkErr != nil {
-		return nil, checkErr
-	}
-	// If mount or symlink doesn't exist, volume reconstruction should be failed
-	if !isExist {
-		return nil, fmt.Errorf("volume: %q is not mounted", uniqueVolumeName)
-	}
-
 	reconstructedVolume := &reconstructedVolume{
```

Comment:
Should we at least check for existence of mount or map path? If they don't exist (but may be pod directory exists) - is there a point in doing reconstruction?

Reply:
Reconstruction is still needed when mount or map path doesn't exist, since we can rely on `unmountDevice` to clean global mount path.

Reply:
but if pod still running, we  add it to asw directly, volume manage will not remount this volume forever, container will use local path instead of  remote path.

Reply:
If Pod is still running, then the volume should exist in DSW, then it won't be added to ASW:https://github.com/kubernetes/kubernetes/blob/e575e60ea42ad58bfe124f44e719e4b7cc1ab2e6/pkg/kubelet/volumemanager/reconciler/reconstruct.go#L87-L97